pros/src/sensors/optical.rs

```diff
@@ -0,0 +1,306 @@
+use core::time::Duration;
+use pros_sys::{PROS_ERR, PROS_ERR_F, OPT_GESTURE_ERR};
+use snafu::Snafu;
+
+use crate::error::{PortError, bail_on, map_errno};
+
+pub const MIN_INTEGRATION_TIME: Duration = Duration::from_millis(3);
+pub const MAX_INTEGRATION_TIME: Duration = Duration::from_millis(712);
+
+pub const MAX_LED_PWM: u8 = 100;
+
+#[derive(Debug)]
+pub struct OpticalSensor {
+    port: u8,
+    gesture_detection_enabled: bool,
+}
+
+impl OpticalSensor {
+    pub fn new(port: u8, gesture_detection_enabled: bool) -> Result<Self, OpticalError> {
```

Comment:
You should add some docs here. You can talk about how leaving gesture detection on when you don't need it is bad for battery life.

Reply:
It's currently unclear if that's true (just a theory that I had), so it'd probably be a good idea to wait until I have a concrete answer from jpearman before I add that detail. I'll add a basic description for now though.

Reply:
Sounds good.